src/main/java/org/nypl/findawayclientlibrary/PlayBookFragment.java

```diff
@@ -359,8 +260,117 @@ public void loadCoverImage() {
+      return;
+    }
+  }
+
+
+  /**
+   * Update the progress bar to reflect where we are in the downloading.
+   * While both the primary and secondary progress percentages are being passed in,
+   * ignore the secondary, and only update the primary for now.  Updating the chapters
+   * is happening too much and thrashing the UI.  To use the secondaryProgress as per chapter
+   * download indicators, will want to take the UI update onto a separate thread.
+   *
+   * @param primaryProgress Represents percent of total book download.
+   * @param secondaryProgress Represents percent of currently downloading chapter.
+   */
+  public void redrawDownloadProgress(Integer primaryProgress, Integer secondaryProgress) {
+    downloadProgress.setProgress(primaryProgress);
+    // downloadProgress.setSecondaryProgress(secondaryProgress);
+
+    // textview feedback of percentages will either go away completely, or move to another location in UI
+    //contentPercentage.setText(getString(R.string.contentPercentage, primaryProgress));
+    //chapterPercentage.setText(getString(R.string.chapterPercentage, secondaryProgress));
+  }
+
+
+  public void resetDownloadProgress() {
+    this.redrawDownloadProgress(0, 0);
+  }
+  /* ------------------------------------ /DOWNLOAD EVENT HANDLERS ------------------------------------- */
+
+
+
   /* ------------------------------------ PLAYBACK EVENT HANDLERS ------------------------------------- */
 
+  /**
+   * Change message and pic on play button to match current state.
+   *
+   * @param newText
+   * @param newImageId
+   */
+  public void redrawPlayButton(String newText, int newImageId) {
+    // TODO: check that code works on kitkat.  if yes, then don't need the "if (checkAndroidVersion() < Build.VERSION_CODES.LOLLIPOP) {"
+    // line in audiobookplaylibrary/PlayBookFragment.  if no, then need that line here.
+    playButton.setImageResource(newImageId);
+    playButton.setTag(newText);
+  }
+
+
+  /**
+   * Move the seek bar to reflect the current playback position within the book chapter.
+   * Making sure the passed event is of type progress update happens in the calling code.
+   *
+   * @param duration
+   * @param position
+   */
+  public void redrawPlaybackPosition(Long duration, Long position) {
+    // TODO: intValue may overflow, and 0 may not be best default value.  fix to handle.
+    int max = duration != null ? duration.intValue() : 0;
+    int progress = position != null ? position.intValue() : 0;
+    playbackSeekBar.setMax(max);
+    playbackSeekBar.setProgress(progress);
+
+    currentTime.setText(DateTimeUtil.millisToHumanReadable(progress));
+    remainingTime.setText(DateTimeUtil.millisToHumanReadable(max - progress));
+  }
+
+
+  /**
+   * Move the audio playback position in response to user shifting the seek bar.
+   *
+   * @param seekBar
+   * @param progress
+   * @param fromUser
+   */
+  public void redrawPlaybackPosition(SeekBar seekBar, int progress, boolean fromUser) {
```

Comment:
Is it correct for this method to be `public`? The same goes for other definitions in this class as well.

Reply:
yep.  they get called from other classes.